In [ ]:
import pandas as pd
import numpy as np

In [ ]:
!gdown --id 1v_GgS4EupZT8pkF47Jf7C4MrSxuhgQ3H

/usr/local/lib/python3.10/dist-packages/gdown/cli.py:121: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  warnings.warn(
Downloading...
From: https://drive.google.com/uc?id=1v_GgS4EupZT8pkF47Jf7C4MrSxuhgQ3H
To: /content/properties_and_hotels_bd.csv
100% 47.7k/47.7k [00:00<00:00, 94.2MB/s]


In [ ]:
df=pd.read_csv("/content/properties_and_hotels_bd.csv")
df.head(10)

name        city      price  stars  \
0    Jatra Flagship Khulna City Centre      Khulna  BDT 3,756      4   
1  Jatra Flagship Rajshahi City Centre    Rājshāhi  BDT 2,485      4   
2                    Hotel Khan Palace     Kuākāta  BDT 4,984      3   
3                    Hotel Orchid Blue       Ināni  BDT 2,547      3   
4                   The Grand Hotel BD      Sylhet  BDT 1,196      3   
5               Tiger Garden Int Hotel      Khulna  BDT 4,208      3   
6                Hotel Noorjahan Grand      Sylhet  BDT 4,128      4   
7                           DESHA TARC     Kushtia  BDT 3,987      4   
8                    Tea Heaven Resort  Sreemangal  BDT 2,769      3   
9     Hotel Grand Brahmaputra Chilmari    Chilmāri  BDT 1,661      5   

    rating_text num_reviews  total_rating  staff  facilities  cleanliness  \
0     Wonderful         242           9.0    8.9         9.5          9.4   
1     Very Good         223           8.4    8.8         8.8          8.5   
2          Good          79           7.8    8.6         8.5          8.1   
3          Good           8           7.7    7.9         8.2          8.6   
4          Good         930           7.5    8.4         7.7          7.8   
5  Review score          89           6.1    7.0         6.8          6.3   
6          Good         965           7.8   -1.0         8.6          8.1   
7  Review score           6           6.3    8.3         5.4          7.9   
8  Review score          17           5.5    6.5         6.6          6.5   
9  Review score           0           0.0   -1.0        -1.0         -1.0   

   comfort  value_for_money  location_rating  
0      8.9              8.8             -1.0  
1      8.5             -1.0             -1.0  
2      8.1              8.1             -1.0  
3     -1.0             -1.0             -1.0  
4      7.6              7.9              8.8  
5      8.2              7.7             -1.0  
6      8.1              8.1              8.1  
7     -1.0             -1.0             -1.0  
8      7.8             -1.0             -1.0  
9     -1.0             -1.0             -1.0

In [ ]:
df.shape

(550, 13)

In [ ]:
df.iloc[:25]

name            city  \
0                   Jatra Flagship Khulna City Centre          Khulna   
1                 Jatra Flagship Rajshahi City Centre        Rājshāhi   
2                                   Hotel Khan Palace         Kuākāta   
3                                   Hotel Orchid Blue           Ināni   
4                                  The Grand Hotel BD          Sylhet   
5                              Tiger Garden Int Hotel          Khulna   
6                               Hotel Noorjahan Grand          Sylhet   
7                                          DESHA TARC         Kushtia   
8                                   Tea Heaven Resort      Sreemangal   
9                    Hotel Grand Brahmaputra Chilmari        Chilmāri   
10                        Aduri Kunjo Resort, Gazipur        Sonātala   
11                         Hotel Omni Residency Dhaka  Gulshan, Dhaka   
12                                 Little Rangpur Inn         Rangpur   
13                       Hotel Bonolota international        Rājshāhi   
14                         Nascent Gardenia Residence           Dhaka   
15                            Rose Garden Guest House          Khulna   
16  Brisa Marina CBC Resort ব্রিসা মেরিনা সিবিসি র...         Patenga   
17                          Sawpnorajjo Park & Resort         Kuākāta   
18                                           Hotel 71           Dhaka   
19                                    Royal Raj Hotel        Rājshāhi   
20     Green Leaf Guest House ColIege Road Sreemongal      Sreemangal   
21                                              Ripon          Khulna   
22                                Hotel Pushpo Bilash          Khulna   
23                               Hotel Shuktara Dhaka           Dhaka   
24                              Hotel Hill Ambassador       Rāngāmāti   

        price  stars   rating_text num_reviews  total_rating  staff  \
0   BDT 3,756      4     Wonderful         242           9.0    8.9   
1   BDT 2,485      4     Very Good         223           8.4    8.8   
2   BDT 4,984      3          Good          79           7.8    8.6   
3   BDT 2,547      3          Good           8           7.7    7.9   
4   BDT 1,196      3          Good         930           7.5    8.4   
5   BDT 4,208      3  Review score          89           6.1    7.0   
6   BDT 4,128      4          Good         965           7.8   -1.0   
7   BDT 3,987      4  Review score           6           6.3    8.3   
8   BDT 2,769      3  Review score          17           5.5    6.5   
9   BDT 1,661      5  Review score           0           0.0   -1.0   
10  BDT 3,876      0  Review score           0           0.0   -1.0   
11  BDT 8,880      4     Very Good         265           8.3   -1.0   
12  BDT 3,322      0          Good         110           7.6    7.9   
13    BDT 941      2  Review score          90           5.4    6.0   
14  BDT 4,568      4     Very Good         151           8.3   -1.0   
15  BDT 1,495      0  Review score          41           6.3    7.1   
16  BDT 6,866      2     Excellent          11           8.6    9.5   
17  BDT 1,661      1  Review score          29           6.1    6.0   
18  BDT 3,502      3          Good       1,417           7.8   -1.0   
19  BDT 5,980      4     Excellent          31           8.6    8.5   
20  BDT 1,329      0     Very Good           2           8.0    7.5   
21    BDT 199      0          Good          17           7.0    6.6   
22  BDT 2,871      3  Review score          20           6.4    7.4   
23  BDT 3,773      3     Very Good         609           8.2   -1.0   
24  BDT 2,658      2  Review score           3           5.3    6.7   

    facilities  cleanliness  comfort  value_for_money  location_rating  
0          9.5          9.4      8.9              8.8             -1.0  
1          8.8          8.5      8.5             -1.0             -1.0  
2          8.5          8.1      8.1              8.1             -1.0  
3          

In [ ]:
#Find out how many duplicate rows are in the data
len(df[df.duplicated()])

48

In [ ]:
df[df['name']=="MD TAREQ AZIZ SHAON"]

name       city      price  stars   rating_text  \
49  MD TAREQ AZIZ SHAON  Atraighat  BDT 1,107      0  Review score   
50  MD TAREQ AZIZ SHAON  Atraighat  BDT 1,107      0  Review score   

   num_reviews  total_rating  staff  facilities  cleanliness  comfort  \
49           0           0.0   -1.0        -1.0         -1.0     -1.0   
50           0           0.0   -1.0        -1.0         -1.0     -1.0   

    value_for_money  location_rating  
49             -1.0             -1.0  
50             -1.0             -1.0

In [ ]:
#Remove the duplicate rows from  the data
df=df.drop_duplicates()

In [ ]:
df.shape

(502, 13)

In [ ]:
#Finding if there are any null values
df.isnull().values.any()

False

In [ ]:
#Finding if there are any missing values
df.isnull().sum()

name               0
city               0
price              0
stars              0
rating_text        0
num_reviews        0
total_rating       0
staff              0
facilities         0
cleanliness        0
comfort            0
value_for_money    0
location_rating    0
dtype: int64

In [ ]:
#Replace rating values where it is written 'Review score' with the text 'None'
df['rating_text'].replace('Review score','None',inplace=True)
df

name         city       price  stars  \
0      Jatra Flagship Khulna City Centre       Khulna   BDT 3,756      4   
1    Jatra Flagship Rajshahi City Centre     Rājshāhi   BDT 2,485      4   
2                      Hotel Khan Palace      Kuākāta   BDT 4,984      3   
3                      Hotel Orchid Blue        Ināni   BDT 2,547      3   
4                     The Grand Hotel BD       Sylhet   BDT 1,196      3   
..                                   ...          ...         ...    ...   
545                          Guest House        Dhaka  BDT 33,225      0   
546                   Kharpara mirabazar       Sylhet   BDT 1,772      3   
547            Hotel Oasis International  Cox's Bazar   BDT 3,544      0   
548                      Hotel Esplanade  Cox's Bazar   BDT 3,322      0   
549                       Cox Hill Tower  Cox's Bazar   BDT 4,264      3   

    rating_text num_reviews  total_rating  staff  facilities  cleanliness  \
0     Wonderful         242           9.0    8.9         9.5          9.4   
1     Very Good         223           8.4    8.8         8.8          8.5   
2          Good          79           7.8    8.6         8.5          8.1   
3          Good           8           7.7    7.9         8.2          8.6   
4          Good         930           7.5    8.4         7.7          7.8   
..          ...         ...           ...    ...         ...          ...   
545        None           0           0.0   -1.0        -1.0         -1.0   
546        None           0           0.0   -1.0        -1.0         -1.0   
547        None           5           3.6    6.0        -1.0         -1.0   
548        None           0           0.0   -1.0        -1.0         -1.0   
549        None           4           3.5    3.8         3.1          5.0   

     comfort  value_for_money  location_rating  
0        8.9              8.8             -1.0  
1        8.5             -1.0             -1.0  
2        8.1              8.1             -1.0  
3       -1.0             -1.0             -1.0  
4        7.6              7.9              8.8  
..       ...              ...              ...  
545     -1.0             -1.0             -1.0  
546     -1.0             -1.0             -1.0  
547     -1.0             -1.0             -1.0  
548     -1.0             -1.0             -1.0  
549      6.9             -1.0             -1.0  

[502 rows x 13 columns]

In [ ]:
#Find out the data types for each column in the dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 502 entries, 0 to 549
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             502 non-null    object 
 1   city             502 non-null    object 
 2   price            502 non-null    object 
 3   stars            502 non-null    int64  
 4   rating_text      502 non-null    object 
 5   num_reviews      502 non-null    object 
 6   total_rating     502 non-null    float64
 7   staff            502 non-null    float64
 8   facilities       502 non-null    float64
 9   cleanliness      502 non-null    float64
 10  comfort          502 non-null    float64
 11  value_for_money  502 non-null    float64
 12  location_rating  502 non-null    float64
dtypes: float64(7), int64(1), object(5)
memory usage: 54.9+ KB


We can see that the 'price' and 'num_reviews' columns are considered as text values here, instead of numeric. This is because of the presence of the comma character to separate values that are 1000 or more. Moreover, the 'price' column consist of the text BDT before each number. So we need to convert these to pure numbers

In [ ]:
df[df['num_reviews']=="1,417"]

Empty DataFrame
Columns: [name, city, price, stars, rating_text, num_reviews, total_rating, staff, facilities, cleanliness, comfort, value_for_money, location_rating]
Index: []

In [ ]:
#Convert num_reviews and price to numeric values
df['num_reviews'] = df['num_reviews'].str.replace(',', '')
df['num_reviews'] = df['num_reviews'].astype(int)
df['price'] = df['price'].str.replace(',', '')
df['price'] = df['price'].str.replace('BDT ', '')
df['price'] = df['price'].astype(int)
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 502 entries, 0 to 549
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             502 non-null    object 
 1   city             502 non-null    object 
 2   price            502 non-null    int64  
 3   stars            502 non-null    int64  
 4   rating_text      502 non-null    object 
 5   num_reviews      502 non-null    int64  
 6   total_rating     502 non-null    float64
 7   staff            502 non-null    float64
 8   facilities       502 non-null    float64
 9   cleanliness      502 non-null    float64
 10  comfort          502 non-null    float64
 11  value_for_money  502 non-null    float64
 12  location_rating  502 non-null    float64
dtypes: float64(7), int64(3), object(3)
memory usage: 54.9+ KB


In [ ]:
df.iloc[:25]

name            city  price  \
0                   Jatra Flagship Khulna City Centre          Khulna   3756   
1                 Jatra Flagship Rajshahi City Centre        Rājshāhi   2485   
2                                   Hotel Khan Palace         Kuākāta   4984   
3                                   Hotel Orchid Blue           Ināni   2547   
4                                  The Grand Hotel BD          Sylhet   1196   
5                              Tiger Garden Int Hotel          Khulna   4208   
6                               Hotel Noorjahan Grand          Sylhet   4128   
7                                          DESHA TARC         Kushtia   3987   
8                                   Tea Heaven Resort      Sreemangal   2769   
9                    Hotel Grand Brahmaputra Chilmari        Chilmāri   1661   
10                        Aduri Kunjo Resort, Gazipur        Sonātala   3876   
11                         Hotel Omni Residency Dhaka  Gulshan, Dhaka   8880   
12                                 Little Rangpur Inn         Rangpur   3322   
13                       Hotel Bonolota international        Rājshāhi    941   
14                         Nascent Gardenia Residence           Dhaka   4568   
15                            Rose Garden Guest House          Khulna   1495   
16  Brisa Marina CBC Resort ব্রিসা মেরিনা সিবিসি র...         Patenga   6866   
17                          Sawpnorajjo Park & Resort         Kuākāta   1661   
18                                           Hotel 71           Dhaka   3502   
19                                    Royal Raj Hotel        Rājshāhi   5980   
20     Green Leaf Guest House ColIege Road Sreemongal      Sreemangal   1329   
21                                              Ripon          Khulna    199   
22                                Hotel Pushpo Bilash          Khulna   2871   
23                               Hotel Shuktara Dhaka           Dhaka   3773   
24                              Hotel Hill Ambassador       Rāngāmāti   2658   

    stars rating_text  num_reviews  total_rating  staff  facilities  \
0       4   Wonderful          242           9.0    8.9         9.5   
1       4   Very Good          223           8.4    8.8         8.8   
2       3        Good           79           7.8    8.6         8.5   
3       3        Good            8           7.7    7.9         8.2   
4       3        Good          930           7.5    8.4         7.7   
5       3        None           89           6.1    7.0         6.8   
6       4        Good          965           7.8   -1.0         8.6   
7       4        None            6           6.3    8.3         5.4   
8       3        None           17           5.5    6.5         6.6   
9       5        None            0           0.0   -1.0        -1.0   
10      0        None            0           0.0   -1.0        -1.0   
11      4   Very Good          265           8.3   -1.0         8.9   
12      0        Good          110           7.6    7.9         8.3   
13      2        None           90           5.4    6.0         5.9   
14      4   Very Good          151           8.3   -1.0         9.1   
15      0        None           41           6.3    7.1         7.1   
16      2   Excellent           11           8.6    9.5         9.8   
17      1        None           29           6.1    6.0         6.3   
18      3        Good         1417           7.8   -1.0         8.4   
19      4   Excellent           31           8.6    8.5         9.2   
20      0   Very Good            2           8.0    7.5         8.8   
21      0        Good           17           7.0    6.6        -1.0   
22      3        None           20           6.4    7.4         6.8   
23      3   Very Good          609           8.2   -1.0         8.8   
24      2        None            3           5.3    6.7         5.0   

    cleanliness  comfort  value_for_money  location_rating  
0           9.4      8.9              8.8             -1.0  

In [ ]:
#Find hotels that have number of reviews less than or equal to 5
df[df['num_reviews']<=10]

name         city  price  \
3                                 Hotel Orchid Blue        Ināni   2547   
7                                        DESHA TARC      Kushtia   3987   
9                  Hotel Grand Brahmaputra Chilmari     Chilmāri   1661   
10                      Aduri Kunjo Resort, Gazipur     Sonātala   3876   
20   Green Leaf Guest House ColIege Road Sreemongal   Sreemangal   1329   
..                                              ...          ...    ...   
545                                     Guest House        Dhaka  33225   
546                              Kharpara mirabazar       Sylhet   1772   
547                       Hotel Oasis International  Cox's Bazar   3544   
548                                 Hotel Esplanade  Cox's Bazar   3322   
549                                  Cox Hill Tower  Cox's Bazar   4264   

     stars rating_text  num_reviews  total_rating  staff  facilities  \
3        3        Good            8           7.7    7.9         8.2   
7        4        None            6           6.3    8.3         5.4   
9        5        None            0           0.0   -1.0        -1.0   
10       0        None            0           0.0   -1.0        -1.0   
20       0   Very Good            2           8.0    7.5         8.8   
..     ...         ...          ...           ...    ...         ...   
545      0        None            0           0.0   -1.0        -1.0   
546      3        None            0           0.0   -1.0        -1.0   
547      0        None            5           3.6    6.0        -1.0   
548      0        None            0           0.0   -1.0        -1.0   
549      3        None            4           3.5    3.8         3.1   

     cleanliness  comfort  value_for_money  location_rating  
3            8.6     -1.0             -1.0             -1.0  
7            7.9     -1.0             -1.0             -1.0  
9           -1.0     -1.0             -1.0             -1.0  
10          -1.0     -1.0             -1.0             -1.0  
20          10.0     -1.0             -1.0             -1.0  
..           ...      ...              ...              ...  
545         -1.0     -1.0             -1.0             -1.0  
546         -1.0     -1.0             -1.0             -1.0  
547         -1.0     -1.0             -1.0             -1.0  
548         -1.0     -1.0             -1.0             -1.0  
549          5.0      6.9             -1.0             -1.0  

[285 rows x 13 columns]

In [ ]:
#Drop the hotels that have number of reviews less than or equal to 5
df.drop(df[df['num_reviews'] <= 10].index, inplace = True)
df.shape

(217, 13)

In [ ]:
#Replace rating values where it is -1.0 with Nan values, since they just mean that there is no rating
df['staff'].replace(-1.0,np.nan,inplace=True)
df['facilities'].replace(-1.0,np.nan,inplace=True)
df['cleanliness'].replace(-1.0,np.nan,inplace=True)
df['comfort'].replace(-1.0,np.nan,inplace=True)
df['value_for_money'].replace(-1.0,np.nan,inplace=True)
df['location_rating'].replace(-1.0,np.nan,inplace=True)
df

name            city  price  stars  \
0      Jatra Flagship Khulna City Centre          Khulna   3756      4   
1    Jatra Flagship Rajshahi City Centre        Rājshāhi   2485      4   
2                      Hotel Khan Palace         Kuākāta   4984      3   
4                     The Grand Hotel BD          Sylhet   1196      3   
5                 Tiger Garden Int Hotel          Khulna   4208      3   
..                                   ...             ...    ...    ...   
467                   Elite Cox's Resort     Cox's Bazar   1550      0   
480                   Grand Beach Resort     Cox's Bazar   1661      0   
484            THE AVENUE HOTEL & SUITES      Chittagong   7309      3   
487                    Quality Inn Dhaka  Gulshan, Dhaka   3667      3   
493                     Royal Palm Hotel          Sylhet   3544      0   

    rating_text  num_reviews  total_rating  staff  facilities  cleanliness  \
0     Wonderful          242           9.0    8.9         9.5          9.4   
1     Very Good          223           8.4    8.8         8.8          8.5   
2          Good           79           7.8    8.6         8.5          8.1   
4          Good          930           7.5    8.4         7.7          7.8   
5          None           89           6.1    7.0         6.8          6.3   
..          ...          ...           ...    ...         ...          ...   
467        None           26           6.6    6.9         NaN          NaN   
480        None           33           4.7    5.2         6.3          5.0   
484        None           18           5.6    6.5         7.2          5.3   
487        None           28           5.3    6.4         6.6          6.3   
493        None           33           6.4    7.2         8.1          7.5   

     comfort  value_for_money  location_rating  
0        8.9              8.8              NaN  
1        8.5              NaN              NaN  
2        8.1              8.1              NaN  
4        7.6              7.9              8.8  
5        8.2              7.7              NaN  
..       ...              ...              ...  
467      NaN              NaN              NaN  
480      NaN              NaN              NaN  
484      7.1              7.0              NaN  
487      5.7              7.6              5.8  
493      NaN              NaN              NaN  

[217 rows x 13 columns]

In [ ]:
#Let's take a look at counts for different types of ratings
df['rating_text'].value_counts()

None           102
Good            75
Very Good       29
Excellent        7
Wonderful        3
Exceptional      1
Name: rating_text, dtype: int64

As we can see, there's only 1 hotel that got an 'Exceptional' rating, and 102 hotels have no rating at all!

In [ ]:
df.describe()

price       stars  num_reviews  total_rating       staff  \
count    217.000000  217.000000   217.000000    217.000000  180.000000   
mean    4951.410138    2.382488   107.728111      6.881106    7.271111   
std     4509.862828    1.679470   164.342221      1.175940    1.112272   
min      199.000000    0.000000    11.000000      3.200000    4.500000   
25%     2326.000000    0.000000    22.000000      6.000000    6.475000   
50%     3502.000000    3.000000    37.000000      7.000000    7.300000   
75%     5791.000000    4.000000   131.000000      7.800000    8.100000   
max    33889.000000    5.000000  1417.000000      9.500000    9.600000   

       facilities  cleanliness     comfort  value_for_money  location_rating  
count  200.000000   196.000000  162.000000       145.000000        92.000000  
mean     7.651500     7.433163    7.827160         7.678621         7.556522  
std      0.970157     1.141601    0.932773         1.188322         1.202406  
min      4.900000     2.500000    4.800000         2.500000         2.500000  
25%      6.875000     6.700000    7.325000         7.300000         7.300000  
50%      7.700000     7.650000    7.900000         7.900000         7.800000  
75%      8.300000     8.200000    8.500000         8.500000         8.200000  
max      9.800000    10.000000   10.000000        10.000000        10.000000

In [ ]:
#View the count,mean, sd, min, max, etc for each (numerical) column, by star rating
star_presence=df.groupby('stars')
#star_presence.first()
star_presence.describe()

In [ ]:
# In some places, the location is included along with "Dhaka". To keep things consistent, we want to change these to just "Dhaka"
df["city"] = df['city'].apply(lambda x: x.split(' ')[-1] if ('Dhaka' in x) else x)
df

#df['city']=df['city'].str.split(' ').str[-1]
#df
#mystr="Gulshan, Dhaka"
#mystr=mystr.split(' ')
#mystr[-1]

name         city  price  stars  \
0      Jatra Flagship Khulna City Centre       Khulna   3756      4   
1    Jatra Flagship Rajshahi City Centre     Rājshāhi   2485      4   
2                      Hotel Khan Palace      Kuākāta   4984      3   
4                     The Grand Hotel BD       Sylhet   1196      3   
5                 Tiger Garden Int Hotel       Khulna   4208      3   
..                                   ...          ...    ...    ...   
467                   Elite Cox's Resort  Cox's Bazar   1550      0   
480                   Grand Beach Resort  Cox's Bazar   1661      0   
484            THE AVENUE HOTEL & SUITES   Chittagong   7309      3   
487                    Quality Inn Dhaka        Dhaka   3667      3   
493                     Royal Palm Hotel       Sylhet   3544      0   

    rating_text  num_reviews  total_rating  staff  facilities  cleanliness  \
0     Wonderful          242           9.0    8.9         9.5          9.4   
1     Very Good          223           8.4    8.8         8.8          8.5   
2          Good           79           7.8    8.6         8.5          8.1   
4          Good          930           7.5    8.4         7.7          7.8   
5          None           89           6.1    7.0         6.8          6.3   
..          ...          ...           ...    ...         ...          ...   
467        None           26           6.6    6.9         NaN          NaN   
480        None           33           4.7    5.2         6.3          5.0   
484        None           18           5.6    6.5         7.2          5.3   
487        None           28           5.3    6.4         6.6          6.3   
493        None           33           6.4    7.2         8.1          7.5   

     comfort  value_for_money  location_rating  
0        8.9              8.8              NaN  
1        8.5              NaN              NaN  
2        8.1              8.1              NaN  
4        7.6              7.9              8.8  
5        8.2              7.7              NaN  
..       ...              ...              ...  
467      NaN              NaN              NaN  
480      NaN              NaN              NaN  
484      7.1              7.0              NaN  
487      5.7              7.6              5.8  
493      NaN              NaN              NaN  

[217 rows x 13 columns]

In [ ]:
#I also noticed that "Tourist Home" has "Uttara, Abdullahpur" as its location, which also basically Dhaka, so let's change that value too
df['city'].replace("Uttara, Ābdullāpur","Dhaka",inplace=True)
df

name         city  price  stars  \
0      Jatra Flagship Khulna City Centre       Khulna   3756      4   
1    Jatra Flagship Rajshahi City Centre     Rājshāhi   2485      4   
2                      Hotel Khan Palace      Kuākāta   4984      3   
4                     The Grand Hotel BD       Sylhet   1196      3   
5                 Tiger Garden Int Hotel       Khulna   4208      3   
..                                   ...          ...    ...    ...   
467                   Elite Cox's Resort  Cox's Bazar   1550      0   
480                   Grand Beach Resort  Cox's Bazar   1661      0   
484            THE AVENUE HOTEL & SUITES   Chittagong   7309      3   
487                    Quality Inn Dhaka        Dhaka   3667      3   
493                     Royal Palm Hotel       Sylhet   3544      0   

    rating_text  num_reviews  total_rating  staff  facilities  cleanliness  \
0     Wonderful          242           9.0    8.9         9.5          9.4   
1     Very Good          223           8.4    8.8         8.8          8.5   
2          Good           79           7.8    8.6         8.5          8.1   
4          Good          930           7.5    8.4         7.7          7.8   
5          None           89           6.1    7.0         6.8          6.3   
..          ...          ...           ...    ...         ...          ...   
467        None           26           6.6    6.9         NaN          NaN   
480        None           33           4.7    5.2         6.3          5.0   
484        None           18           5.6    6.5         7.2          5.3   
487        None           28           5.3    6.4         6.6          6.3   
493        None           33           6.4    7.2         8.1          7.5   

     comfort  value_for_money  location_rating  
0        8.9              8.8              NaN  
1        8.5              NaN              NaN  
2        8.1              8.1              NaN  
4        7.6              7.9              8.8  
5        8.2              7.7              NaN  
..       ...              ...              ...  
467      NaN              NaN              NaN  
480      NaN              NaN              NaN  
484      7.1              7.0              NaN  
487      5.7              7.6              5.8  
493      NaN              NaN              NaN  

[217 rows x 13 columns]

In [ ]:
city_presence=df.groupby('city')

In [ ]:
city_presence['stars'].mean()

city
Barisāl         3.000000
Bogra           2.500000
Bāndarban       1.500000
Chittagong      1.500000
Cox's Bazar     1.586207
Dhaka           2.760684
Dhoāpālong      0.000000
Jessore         4.000000
Khulna          1.833333
Kuākāta         1.333333
Maulvi Bāzār    3.000000
Patenga         2.000000
Rangpur         0.000000
Rājshāhi        3.250000
Rājārbāg        2.000000
Rāngāmāti       0.000000
Sreemangal      1.714286
Sylhet          2.761905
Āminpur         0.000000
Name: stars, dtype: float64

In [ ]:
city_presence['total_rating'].max()

city
Barisāl         8.0
Bogra           6.5
Bāndarban       5.5
Chittagong      8.4
Cox's Bazar     7.3
Dhaka           9.1
Dhoāpālong      8.3
Jessore         7.5
Khulna          9.0
Kuākāta         8.7
Maulvi Bāzār    6.9
Patenga         8.6
Rangpur         7.6
Rājshāhi        8.6
Rājārbāg        4.4
Rāngāmāti       5.8
Sreemangal      8.1
Sylhet          9.5
Āminpur         6.4
Name: total_rating, dtype: float64

In [ ]:
city_presence['price'].max()

city
Barisāl          6693
Bogra            3101
Bāndarban        3101
Chittagong      14951
Cox's Bazar      6831
Dhaka           33889
Dhoāpālong       4430
Jessore          8838
Khulna           4208
Kuākāta          4984
Maulvi Bāzār     2193
Patenga          6866
Rangpur          3322
Rājshāhi         5980
Rājārbāg         1357
Rāngāmāti        1661
Sreemangal      16059
Sylhet          11629
Āminpur          8648
Name: price, dtype: int64

In [ ]:
#let's see how many different starred hotel each city has:
pd.crosstab(df["city"],df["stars"],margins=True)

stars          0  1   2   3   4   5  All
city                                    
Barisāl        0  0   0   1   0   0    1
Bogra          0  0   1   1   0   0    2
Bāndarban      1  0   0   1   0   0    2
Chittagong     8  1   2   2   2   1   16
Cox's Bazar   14  1   1  11   0   2   29
Dhaka         23  3   8  41  29  13  117
Dhoāpālong     1  0   0   0   0   0    1
Jessore        0  0   0   0   2   0    2
Khulna         2  1   0   2   1   0    6
Kuākāta        1  1   0   1   0   0    3
Maulvi Bāzār   0  0   0   1   0   0    1
Patenga        0  0   1   0   0   0    1
Rangpur        1  0   0   0   0   0    1
Rājshāhi       0  0   1   1   2   0    4
Rājārbāg       0  0   1   0   0   0    1
Rāngāmāti      1  0   0   0   0   0    1
Sreemangal     3  0   0   4   0   0    7
Sylhet         4  0   0  11   5   1   21
Āminpur        1  0   0   0   0   0    1
All           60  7  15  77  41  17  217

In [ ]:
#Finally let's store the cleaned and organized dataframe into a new csv file
df.to_csv('/content/cleaned_dataset_properties_and_hotels.csv', encoding='utf-8', index=False)